redfish-core/include/registries_selector.hpp

```diff
@@ -1,27 +1,94 @@
 #pragma once
+#include "registries.hpp"
 #include "registries/base_message_registry.hpp"
+#include "registries/bios_message_registry.hpp"
+#include "registries/heartbeat_event_message_registry.hpp"
+#include "registries/license_message_registry.hpp"
 #include "registries/openbmc_message_registry.hpp"
+#include "registries/resource_event_message_registry.hpp"
 #include "registries/task_event_message_registry.hpp"
+#include "registries/telemetry_message_registry.hpp"
 
+#include <optional>
 #include <span>
 #include <string_view>
 
 namespace redfish::registries
 {
+struct HeaderAndUrl
+{
+    const Header& header;
+    const char* url;
+};
+
+inline std::optional<registries::HeaderAndUrl>
+    getRegistryHeaderAndUrlFromPrefix(std::string_view registryName)
+{
+    if (base::header.registryPrefix == registryName)
+    {
+        return HeaderAndUrl{base::header, base::url};
+    }
+    if (bios::header.registryPrefix == registryName)
+    {
+        return HeaderAndUrl{bios::header, bios::url};
+    }
+    if (heartbeat_event::header.registryPrefix == registryName)
+    {
+        return HeaderAndUrl{heartbeat_event::header, heartbeat_event::url};
+    }
+    if (license::header.registryPrefix == registryName)
+    {
+        return HeaderAndUrl{license::header, license::url};
+    }
+    if (openbmc::header.registryPrefix == registryName)
+    {
+        return HeaderAndUrl{openbmc::header, openbmc::url};
+    }
+    if (resource_event::header.registryPrefix == registryName)
+    {
+        return HeaderAndUrl{resource_event::header, resource_event::url};
+    }
+    if (task_event::header.registryPrefix == registryName)
+    {
+        return HeaderAndUrl{task_event::header, task_event::url};
+    }
+    if (telemetry::header.registryPrefix == registryName)
+    {
+        return HeaderAndUrl{telemetry::header, telemetry::url};
+    }
+    return std::nullopt;
+}
+
 inline std::span<const MessageEntry>
     getRegistryFromPrefix(std::string_view registryName)
 {
-    if (task_event::header.registryPrefix == registryName)
+    if (base::header.registryPrefix == registryName)
     {
-        return {task_event::registry};
+        return {base::registry};
+    }
+    if (heartbeat_event::header.registryPrefix == registryName)
```

Comment:
Does this function need the bios::header.registryPrefix check also?

Reply:
> Does this function need the bios::header.registryPrefix check also?

The registry messages (e.g. `redfish::registries::task_event::registry` are derived from `/https://redfish.dmtf.org/registries/<registry>.json` like `https://redfish.dmtf.org/registries/TaskEvent.1.0.4.json`.

However, `Bios` registry is not there and thus`registries/bios_message_registry.hpp` is not manually created but without `registry` part.
Also the actual `/redfish/v1/Registries/BiosAttributeRegistry/BiosAttributeRegistry` is handled separately via `getBiosAttributeRegistry()' by shortcutting it.

```
inline void handleMessageRegistryGet(...
    const std::string& registry, const std::string& registryMatch)

{
....
    if (registry == "BiosAttributeRegistry" &&
        registryMatch == "BiosAttributeRegistry")
    {
        getBiosAttributeRegistry(asyncResp);  <===
        return;
    }
```

At this time, it is not needed, but I'add an comparison for `bios` and return `{openbmc::registry}` as how the current code behaves.

I think eventually this needs to be re-addressed  (probably as a part of BIOS upstreaming effort).